server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -879,6 +898,13 @@ List<String> getPipelineStack() {
         return pipelineStack;
     }
 
+    /**
+     * @return The access pattern for any currently executing pipelines, or null if no pipelines are in progress for this doc
+     */
+    public IngestPipelineFieldAccessPattern getCurrentAccessPattern() {
```

Comment:
Is it intentional that this method isn't used above (and seems to only be used in the test)?

Reply:
not used above yet. I'm currently using it in a couple places in the next set of changes I'm working on.